### PR TITLE
ref(examples): Refactored example helper functions

### PR DESCRIPTION
**Pull request by kartik4949**
solves

#403

### CODE REVIEW

File: examples/layers/preprocessing/utils.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        resize, img_size=img_size, num_classes=num_classes
+    )
+
+    # Get tf dataset.
+    train_ds = train_ds.map(lambda x, y: resize_partial(x, y)).batch(batch_size)
+    return train_ds
+
+
+def apply_augmentation(ds, augmentation_fxn=None):
+    ds = ds.map(augmentation_fxn, num_parallel_calls=tf.data.AUTOTUNE)
```
Comment (LukeWood) [inline examples/layers/preprocessing/utils.py:50]: no need for this helper, we can just inline ds.map(my_call)

File: examples/layers/preprocessing/utils.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    return ds
+
+
+def visualize_dataset(ds=None, take=1):
+    for images, _ in ds.take(take):
```
Comment (LukeWood) [inline examples/layers/preprocessing/utils.py:55]: we could also do:

```
images, labels = next(iter(ds.take(1)))
```
and drop the take param

**Review by LukeWood — CHANGES_REQUESTED** [review]
Minor comments, thanks!

File: examples/layers/preprocessing/utils.py (hunk 2 above)
Comment (kartik4949) [reply examples/layers/preprocessing/utils.py]: agreed!

File: examples/layers/preprocessing/utils.py (hunk 1 above)
Comment (kartik4949) [reply examples/layers/preprocessing/utils.py]: yeah in inline ds map should be fine, thanks

File: examples/layers/preprocessing/random_hue_demo.py [hunk 3]
```diff
@@ -16,38 +16,23 @@
...
-            plt.subplot(3, 3, i + 1)
-            plt.imshow(images[i].numpy().astype("uint8"))
-            plt.axis("off")
-        plt.show()
+
+    # Apply augmentation.
+    train_ds = train_ds.map(lambda x, y: (random_hue(x), y))
+
+    # visualize.
+    utils.visualize_dataset(train_ds)
```
Comment (kartik4949) [inline examples/layers/preprocessing/random_hue_demo.py:35]: @LukeWood #446

Comment (LukeWood) [reply examples/layers/preprocessing/random_hue_demo.py:35]: Missed this PR, thanks Kartik!  I will be sure to do a pass on this!

**Comment by LukeWood** [comment]
Hi @kartik4949 I have updated the tools and merged this PR.  Please feel free to migrate the rest of the tools to use demo_utils.  I have opened https://github.com/keras-team/keras-cv/issues/485 to track this.  Thank you for your contribution!

**Comment by kartik4949** [comment]
@LukeWood thanks for updating, sure I will, cheers!

**Pull request merged**